x-pack/platform/plugins/shared/fleet/server/tasks/sync_integrations/sync_integrations_on_remote.ts

```diff
@@ -180,3 +221,262 @@ export const syncIntegrationsOnRemote = async (
+
+  try {
+    const installedPipelines = await getPipeline(esClient, CUSTOM_ASSETS_PREFIX, abortController);
+    const installedComponentTemplates = await getComponentTemplate(
+      esClient,
+      CUSTOM_ASSETS_PREFIX,
+      abortController
+    );
+
+    const componentTemplatesByName = (
+      installedComponentTemplates?.component_templates || []
+    ).reduce((acc, componentTemplate) => {
+      if (componentTemplate?.name) {
+        acc[componentTemplate.name] = componentTemplate.component_template?.template;
+      }
+      return acc;
+    }, {} as Record<string, ClusterComponentTemplateSummary>);
+    const componentTemplates = installedComponentTemplates?.component_templates
+      ? componentTemplatesByName
+      : undefined;
+    const result: RemoteSyncedCustomAssetsRecord = {};
+
+    // compare custom pipelines and custom component templates
+    Object.entries(ccrCustomAssets).forEach(([ccrCustomName, ccrCustomAsset]) => {
+      const res = compareCustomAssets({
+        ccrCustomAsset,
+        ingestPipelines: installedPipelines,
+        componentTemplates,
+      });
+      result[ccrCustomName] = res;
+    });
+    return result;
+  } catch (error) {
+    logger.error('error', error.message);
+    return { error };
+  }
+};
+
+const compareCustomAssets = ({
+  ccrCustomAsset,
+  ingestPipelines,
+  componentTemplates,
+}: {
+  ccrCustomAsset: CustomAssetsData;
+  ingestPipelines?: IngestGetPipelineResponse;
+  componentTemplates?: Record<string, ClusterComponentTemplateSummary>;
+}): RemoteSyncedCustomAssetsStatus => {
+  if (ccrCustomAsset.type === 'ingest_pipeline') {
+    if (!ingestPipelines) {
+      return {
+        ...ccrCustomAsset,
+        sync_status: 'failed' as SyncStatus.FAILED,
+        error: `No custom pipelines found on remote index`,
+      };
+    }
+
+    const installedAsset = ingestPipelines[ccrCustomAsset?.name];
+
+    if (isEqual(installedAsset, ccrCustomAsset?.pipeline)) {
+      return {
+        ...ccrCustomAsset,
+        sync_status: 'completed' as SyncStatus.COMPLETED,
+      };
+    } else {
+      return {
+        ...ccrCustomAsset,
+        sync_status: 'synchronizing' as SyncStatus.SYNCHRONIZING,
+      };
+    }
+  } else if (ccrCustomAsset.type === 'component_template') {
+    if (!componentTemplates) {
+      return {
+        ...ccrCustomAsset,
+        sync_status: 'failed' as SyncStatus.FAILED,
+        error: `No component templates found on remote index`,
+      };
+    }
+
+    const installedAsset = componentTemplates[ccrCustomAsset?.name];
+    if (isEqual(installedAsset, ccrCustomAsset?.template)) {
+      return {
+        ...ccrCustomAsset,
+        sync_status: 'completed' as SyncStatus.COMPLETED,
+      };
+    } else {
+      return {
+        ...ccrCustomAsset,
+        sync_status: 'synchronizing' as SyncStatus.SYNCHRONIZING,
+      };
+    }
+  }
+  return {} as RemoteSyncedCustomAssetsStatus;
+};
+
+export const getRemoteSyncedIntegrationsStatus = async (
```

Comment:
I think this function should be moved out to another file, it's a bit confusing to have it next to the task logic.